### PR TITLE
Use casesvc sampleunitid endpoint

### PR DESCRIPTION
**Pull request by hbrace**
# Motivation and Context
The casesvc endpoint that was being used to get the list of cases did not return the expected result when run in preprod. A different endpoint is now being used to get the cases in a single request, rather than a request for each case.

# How to test?
Have EQ running and create a collection exercise. Ensure you start some cases so they are marked as INPROGRESS then run the script. You should see a 200 response in the log for all the cases you have started.

# Links
https://trello.com/c/lNmyBmPH

### CODE REVIEW

**Comment by hbrace** [comment]
[SUS014: Send partial Data Downstream (13)](https://trello.com/c/lNmyBmPH/214-sus014-send-partial-data-downstream-13)

File: scripts/flush.py [hunk 1]
```diff
@@ -41,15 +41,14 @@ def main(collection_ex_id):
 
     samples = [sample['id'] for sample in sample_units]
 
+    sample_return = requests.get(case_url + "sampleunitids?sampleUnitId=" + ','.join(samples), auth=config["CASE_AUTH"])
```
Comment (jcox-dev) [inline scripts/flush.py]: Getting the following error here when running against a 500 row sample: 

```python
requests.exceptions.HTTPError: 414 Client Error: URI Too Long for url: http://localhost:8171/cases/sampleunitids?sampleUnitId=bde74017-fbef-4851-a0cb...[snip]
```

File: scripts/flush.py [hunk 2]
```diff
@@ -41,15 +41,14 @@ def main(collection_ex_id):
...
+    
     case_inprogress = []
-    for sample in samples:
-        sample_return = requests.get(case_url + "?sampleUnitId=" + sample, auth=config["CASE_AUTH"])
-        sample_return.raise_for_status()
-        case_return = sample_return.json()
-        case_return = case_return[0]
-        if case_return["caseGroup"]["collectionExerciseId"] == str(collection_ex[0]) and case_return["caseGroup"][
+    for sample in sample_return.json():
+        if sample["caseGroup"]["collectionExerciseId"] == str(collection_ex[0]) and sample["caseGroup"][
```
Comment (jcox-dev) [inline scripts/flush.py]:
```suggestion
        if (sample["caseGroup"]["collectionExerciseId"] == str(collection_ex[0])
           and sample["caseGroup"]["caseGroupStatus"] == "INPROGRESS"):
            case_inprogress.append(sample)```

File: scripts/flush.py (hunk 1 above)
Comment (jcox-dev) [reply scripts/flush.py]: (created using `make setup`)

File: scripts/flush.py [hunk 3]
```diff
@@ -41,15 +41,14 @@ def main(collection_ex_id):
 
     samples = [sample['id'] for sample in sample_units]
```
Comment (jcox-dev) [inline scripts/flush.py]:
```suggestion
     samples = [sample['id'] for sample in sample_units]
     print(f'Sample units for collection exercise: {len(samples)}')
```

File: scripts/flush.py [hunk 4]
```diff
@@ -41,15 +41,14 @@ def main(collection_ex_id):
...
-        sample_return = requests.get(case_url + "?sampleUnitId=" + sample, auth=config["CASE_AUTH"])
-        sample_return.raise_for_status()
-        case_return = sample_return.json()
-        case_return = case_return[0]
-        if case_return["caseGroup"]["collectionExerciseId"] == str(collection_ex[0]) and case_return["caseGroup"][
+    for sample in sample_return.json():
+        if sample["caseGroup"]["collectionExerciseId"] == str(collection_ex[0]) and sample["caseGroup"][
             "caseGroupStatus"] == "INPROGRESS":
-            case_inprogress.append(case_return)
+            case_inprogress.append(sample)
```
Comment (jcox-dev) [inline scripts/flush.py]:
```suggestion
            case_inprogress.append(sample)
        
        print(f'Cases in progress: {len(case_inprogress)}')
```

File: scripts/flush.py [hunk 5]
```diff
@@ -121,6 +120,7 @@ def flush_cases(case_id):
         flush_response = requests.post(flush_url)
         print(f'Response from flushing {case_id}: {flush_response.status_code}')
 
+
 if __name__ == '__main__':
     collection_ex = sys.argv[1:]
```
Comment (jcox-dev) [inline scripts/flush.py]:
```suggestion
     print(f'Using {config_info.__name__}')
     collection_ex = sys.argv[1:]
```

File: scripts/flush.py [hunk 6]
```diff
@@ -121,6 +120,7 @@ def flush_cases(case_id):
...
     print("Collection exercise: " + str(collection_ex))
```
Comment (jcox-dev) [inline scripts/flush.py]:
```suggestion
    print(f'Collection exercise: {str(collection_ex)}')
```

File: scripts/flush.py (hunk 3 above)
Comment (hbrace) [reply scripts/flush.py]: Added in along with the other suggestions 👍

I'd expect 404 where multiple IACs are used for a case, the first request with an IAC will be used to flush the session from EQ then the following requests will return 404s as that case will have been flushed. Dave was speaking to EQ last week about EQ not starting a new session for each IAC (not sure if this is the case in preprod/prod).

File: scripts/flush.py (hunk 1 above)
Comment (hbrace) [reply scripts/flush.py]: The requests to the casesvc/sampleunitid endpoint are now being sent in chunks of 10.

File: scripts/flush.py (hunk 2 above)
Comment (hbrace) [reply scripts/flush.py]: 👍

File: scripts/flush.py [hunk 7]
```diff
@@ -40,17 +40,27 @@ def main(collection_ex_id):
...
-    sample_return = requests.get(case_url + "sampleunitids?sampleUnitId=" + ','.join(samples), auth=config["CASE_AUTH"])
-    sample_return.raise_for_status()
-    
     case_inprogress = []
-    for sample in sample_return.json():
-        if sample["caseGroup"]["collectionExerciseId"] == str(collection_ex[0]) and sample["caseGroup"][
-            "caseGroupStatus"] == "INPROGRESS":
+    for sample in sample_return:
+        if (sample["caseGroup"]["collectionExerciseId"] == str(collection_ex[0]) and sample["caseGroup"][
+            "caseGroupStatus"] == "INPROGRESS"):
```
Comment (jcox-dev) [inline scripts/flush.py]: 🎨 I would avoid continuation like this and try and keep the entire member block in tact: `sample["caseGroup"]["caseGroupStatus"]`

**Comment by jcox-dev** [comment]
Still getting some 404s 🤔

**Pull request merged**